### PR TITLE
React app setup done with Webpack, Babel, SASS

### PR DESCRIPTION
**Pull request by akshayp-webonise**
React app setup done with Webpack, Babel, SASS

### CODE REVIEW

File: react-app/webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      {
+        test: /\.sass$/,
+        use: ExtractTextPlugin.extract({
+          fallback: 'style-loader',
+          use: ['css-loader', 'sass-loader']
+        })
+      }
+    ]
+  },
+  resolve: {
```
Comment (swapnil-webonise) [inline react-app/webpack.config.js]: Try webpack alias for file import

File: react-app/webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
+const webpack = require('webpack');
```
Comment (swapnil-webonise) [inline react-app/webpack.config.js]: Use babel-register, for writing es6 code here

**Review by swapnil-webonise — CHANGES_REQUESTED** [review]

File: react-app/package.json [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  ],
+  "author": "Akshay Pawar",
+  "license": "ISC",
+  "devDependencies": {
+    "babel-core": "^6.26.0",
+    "babel-loader": "^7.1.4",
+    "babel-preset-env": "^1.6.1",
+    "babel-preset-react": "^6.24.1",
+    "css-loader": "^0.28.11",
+    "extract-text-webpack-plugin": "^4.0.0-beta.0",
```
Comment (abhinav-webonise) [inline react-app/package.json]: You are using the beta version of this npm package which is very dangerous. I would suggest you go with last stable version.

File: react-app/package.json [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  "author": "Akshay Pawar",
+  "license": "ISC",
+  "devDependencies": {
+    "babel-core": "^6.26.0",
+    "babel-loader": "^7.1.4",
+    "babel-preset-env": "^1.6.1",
+    "babel-preset-react": "^6.24.1",
+    "babel-register": "^6.26.0",
+    "css-loader": "^0.28.11",
+    "extract-text-webpack-plugin": "^4.0.0-beta.0",
```
Comment (akshayp-webonise) [inline react-app/package.json]: @swapnil-webonise @abhinav-webonise  please suggest a solution here, the stable package itself doesn't provide support for Webpack 4, but provides support in Beta version.
https://github.com/webpack-contrib/extract-text-webpack-plugin/issues/701
However this plugin is featured in Webpack Docs and is widely used. 
https://webpack.js.org/plugins/extract-text-webpack-plugin/
Should we stop using this?

Comment (abhinav-webonise) [reply react-app/package.json]: I agree with your point, but for live projects, It is not recommended to use alpha or beta versions of packages If you are doing this up for sample apps or just to practice, go ahead.
Basically, a **beta version** is a version of a piece of software that is made available for testing, typically by a limited number of users outside the company that is developing it, before its general release. 

FYI **Beta Versions** aren't stable.

Comment (akshayp-webonise) [reply react-app/package.json]: I still don't see a concrete solution here. Please suggest how should we solve such a problem where we aren't able to extract css to a file, along with SASS in it. Also there's a high possibility that we might use Webpack 4 in the Iconic rebuild, so this problem is going to arise there also, it's better we solve it here now.

Comment (abhinav-webonise) [reply react-app/package.json]: use webpack 3 Stable supporting version, until the time they aren't supporting or find an alternative to your beta package.

Comment (akshayp-webonise) [reply react-app/package.json]: Solved. Used another plugin named `mini-css-extract-plugin`, https://github.com/webpack-contrib/mini-css-extract-plugin. It supports extraction of CSS to a seperate file and also SASS.

File: react-app/src/app.js [hunk 5]
```diff
@@ -0,0 +1,2 @@
+const reactApp = require('./components/HelloMessage.jsx');
```
Comment (owais-webonise) [inline react-app/src/app.js]: Use import which is a part of ES6, avoid using require

File: react-app/src/components/HelloMessage.jsx [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import ReactDOM from 'react-dom';
+import TestChildComponent from 'Components/TestChildComponent';
+
+class HelloMessage extends React.Component {
```
Comment (owais-webonise) [inline react-app/src/components/HelloMessage.jsx]: Use class name properly, you can use 'HelloComponent' instead

File: react-app/webpack.config.babel.js [hunk 7]
```diff
@@ -0,0 +1,67 @@
+import webpack from 'webpack';
+import path from 'path';
+import HtmlWebpackPlugin from 'html-webpack-plugin';
+import MiniCssExtractPlugin from 'mini-css-extract-plugin';
+
+module.exports = {
+  entry: ['./src/app.js'],
```
Comment (owais-webonise) [inline react-app/webpack.config.babel.js]: Try to add a root path(parent folder) here, which you can use also for later purposes in webpack.

**Review by owais-webonise — CHANGES_REQUESTED** [review]

**Review by swapnil-webonise — APPROVED** [review]

**Review by owais-webonise — APPROVED** [review]

**Pull request merged**